### DISCUSSION TITLE
start failed after patronictl restart postgres-cluster

### DISCUSSION THREAD

**Discussion by jetchan88** [autobase-tech/autobase#676] [Q&A]
```
+ Cluster: postgres-cluster (7307507502278680192) ------+----+-----------+
| Member      | Host           | Role    | State        | TL | Lag in MB |
+-------------+----------------+---------+--------------+----+-----------+
| alidbsdev01 | 192.168.172.68 | Replica | start failed |    |   unknown |
| alidbsdev02 | 192.168.172.69 | Replica | start failed |    |   unknown |
| alidbsdev03 | 192.168.172.70 | Replica | start failed |    |   unknown |
+-------------+----------------+---------+--------------+----+-----------+
```

```
Jun 14 09:47:47 alidbsdev03 patroni[1302]: 2024-06-14 09:47:47,486 WARNING: Postgresql is not running.
Jun 14 09:47:47 alidbsdev03 patroni[1302]: 2024-06-14 09:47:47,486 INFO: Lock owner: None; I am alidbsdev03
Jun 14 09:47:47 alidbsdev03 patroni[1302]: 2024-06-14 09:47:47,488 INFO: pg_controldata:
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   pg_control version number: 1300
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Catalog version number: 202209061
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Database system identifier: 7307507502278680192
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Database cluster state: shut down in recovery
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   pg_control last modified: Fri Jun 14 09:12:26 2024
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint location: C41/EE000028
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's REDO location: C41/EE000028
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's REDO WAL file: 0000002D00000C41000000EE
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's TimeLineID: 45
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's PrevTimeLineID: 45
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's full_page_writes: on
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's NextXID: 0:12775692
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's NextOID: 1784167
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's NextMultiXactId: 4
Jun 14 09:47:47 alidbsdev03 patroni[1302]:   Latest checkpoint's NextMultiOffset: 7
```

**Comment by jetchan88** [comment]
After change value for work_mem  and shared_buffers. It can start now. but still got so OOM

```
2024-06-14 10:04:52 HKT [37465-1]  ERROR:  out of memory
2024-06-14 10:04:52 HKT [37465-2]  DETAIL:  Failed on request of size 1073741822 in memory context "TopTransactionContext".
2024-06-14 10:04:52 HKT [37465-3]  CONTEXT:  automatic vacuum of table "c_MATCH"
```

**Comment by vitabaks** [comment]
Hi @jetchan88 

This issue is not related to cluster deployment but to maintenance issues. Consider [sponsorship](https://github.com/vitabaks/postgresql_cluster?tab=readme-ov-file#sponsor-this-project), some of the subscriptions include individual support.

Where to start:

1. Check the monitoring system to determine the reason for the increased memory consumption. For example: more connections or a change in workload. Which processes consume memory.
2. If query optimization is not considered, then try to reduce memory usage (shared_buffers, work_mem, maintenance_work_mem) or increase the resources of the database server.
3. Monitor database performance metrics when changing parameters. If the memory usage parameters are reduced, performance degradation is possible.

**Comment by jetchan88** [comment]
Hi Vitaliy Kukharik,
We need to buy postgresql DB maintenance support from you.
But we have to follow the our company policy to apply budge to pay.
I wonder if we could  pay quarterly or yearly?

for the maintenance support , becoming the small sponsor,  could you please cover the following items
1. Backup/Restore the PG cluster
2. Backup/Restore the PG data
3. Problem fixing i.e  (Replication, Adding replica node, performance tuning)

Best RegardsJet    On Friday, June 14, 2024 at 05:03:23 PM GMT+8, Vitaliy Kukharik ***@***.***> wrote:  
 
 


Hi @jetchan88

This issue is not related to cluster deployment but to maintenance issues. Consider sponsorship, some of the subscriptions include individual support.

Where to start:
   
   - Check the monitoring system to determine the reason for the increased memory consumption. For example: more connections or a change in workload. Which processes consume memory.
   - If query optimization is not considered, then try to reduce memory usage (shared_buffers, work_mem, maintenance_work_mem) or increase the resources of the database server.
   - Monitor database performance metrics when changing parameters. If the memory usage parameters are reduced, performance degradation is possible.

**Comment by vitabaks** [comment]
Hi Jet,

Thank you for reaching out.

At present, we have a sponsorship programs that includes maintenance support services, with payments typically handled on a monthly basis.

> for the maintenance support , becoming the small sponsor,  could you please cover the following items

Yes, it is included.

Please note that the “Small Sponsor” program is designed for clients who require infrequent support, with a limited number of inquiries per month and a response time of 12 to 48 hours. If you require more immediate or comprehensive support, I would recommend considering our other programs, such as the “Sponsor” or “Sponsor+” options, which offer faster response times and more extensive support.You can find more details about these programs here https://github.com/sponsors/vitabaks
